src/ui-test/camelProjectOfferingTest.ts

```diff
@@ -0,0 +1,78 @@
+import { QuickOpenBox, InputBox, QuickPickItem } from "vscode-extension-tester";
+
+import { openCommandPrompt, typeCommandConfirm, convertArrayObjectsToText, convertArrayObjectsToTextAndDescription } from "./common/commonUtils";
+
+import { assertEqualOptions } from "./common/testUtils";
+
+import { ProjectInitializer } from "./common/projectInitializerConstants";
+import { Catalog } from "../Catalog";
+import { expect } from "chai";
+
+const CAMEL_MISSIONS_EXPECTED = [
+    'circuit-breaker', 'configmap', 'health-check', 'rest-http', 'istio-distributed-tracing'
+];
+
+/**
+ * @author odockal@redhat.com
+ */
+export function testFuseProjectOffering() {
+    describe('Verify Project initializer Camel/Fuse Command palette options', async function() {
+
+        let inputBox: QuickOpenBox;
+        let catalogFuse: any;
+
+        before(async function() {
+            this.timeout(5000);
+            let catalogBuilder = new Catalog('https://forge.api.openshift.io/api/');
+            let catalog = await catalogBuilder.getCatalog();
+            catalogFuse = filterCatalogForRuntimes(catalog, ProjectInitializer.CAMEL_FUSE_RUNTIME_IDS);
+        });
+
+        it('Available missions and runtime versions correspond to Camel/Fuse project catalog', async function () {
+            this.timeout(30000);
+            for (let mission of catalogFuse.missions) {
+                console.log("\tVerifying runtimes and versions for " + mission.id + " mission");
+                await openCommandPrompt();
+                await typeCommandConfirm('>' + ProjectInitializer.PI_GENERAL.camel);
+                inputBox = await InputBox.create();
+                await inputBox.selectQuickPick(mission.id);
+                let expectedRefs = findRefsForMissions(mission.id, catalogFuse.boosters);
+                let expected = expectedRefs.map( (runver: any) => runver.runtime + " " + runver.version);
+                inputBox = await InputBox.create();
+                let actual = await convertArrayObjectsToTextAndDescription<QuickPickItem>(await inputBox.getQuickPicks());
+                expect(expected).to.have.same.members(actual);
+                await inputBox.cancel();
+            }
+        });
+
+        it('Verify options available for Camel/Fuse project', async function () {
+            this.timeout(10000);
```

Comment:
Can you try to use a conditional wait?
Have you spotted something missing to use ones?